src/pybind/mgr/dashboard/tools.py

```diff
@@ -35,12 +37,24 @@ def _setup(self):
     def request_begin(self):
         req = cherrypy.request
         user = JwtManager.get_username()
-        if user:
-            logger.debug("[%s:%s] [%s] [%s] %s", req.remote.ip,
-                         req.remote.port, req.method, user, req.path_info)
-        else:
-            logger.debug("[%s:%s] [%s] %s", req.remote.ip,
-                         req.remote.port, req.method, req.path_info)
+        # Log the request.
+        logger.debug('[%s:%s] [%s] [%s] %s', req.remote.ip, req.remote.port,
+                     req.method, user, req.path_info)
+        # Audit the request.
+        if Settings.AUDIT_API_ENABLED and req.method not in ['GET']:
+            url = build_url(req.remote.ip, scheme=req.scheme,
+                            port=req.remote.port)
+            msg = '[DASHBOARD] from=\'{}\' path=\'{}\' method=\'{}\' ' \
+                'user=\'{}\''.format(url, req.path_info, req.method, user)
+            if Settings.AUDIT_API_LOG_PAYLOAD:
+                params = req.params if req.params else {}
+                params.update(get_request_body_params(req))
+                # Hide sensitive data like passwords, ... Extend list if necessary.
+                for param_name in ['password', 'passwd']:
+                    if param_name in params:
+                        params[param_name] = '*' * len(params[param_name])
```

Comment:
I'm slightly concerned about the security of replacing each character of a password with an asterisk. IMO it would be better if the length of a password wasn't logged. Can you please adapt it?

Reply:
Done